keps/sig-network/4444-service-routing-preference/README.md

```diff
@@ -0,0 +1,998 @@
+    route the traffic. 
+  * **Goal:** The aim with Auto was to allow implementations to be as smart as
+    possible and choose what it considered to be the best routing criteria. The
+    field didn’t support any other value by design with the thought that the
+    implementation should be able to make the best possible choice for the user.
+  * **Limitation:** 
+      * While designed for intelligent routing, this mode offers less user
+        control and can be less predictable. Users filed issues reporting that
+        hints weren't being applied or didn't function as expected.
+      * The design attempted to strike a balance between safety and preferential
+        zone-based routing, but may have fallen short in achieving both
+        effectively. Some users prioritize predictability for performance
+        optimization, even when accepting potential endpoint overload in certain
+        situations.
+
+* InternalTrafficPolicy 
+  * The spec.internalTrafficPolicy field with the "Local" setting restricts
+    traffic to endpoints on the same node as the originating Pod. 
+  * **Goal:** With the deprecation of topologyKeys, one of the driving reasons was
+    to satisfy the semantic use cases of strictly routing the traffic to the
+    same node.
+  * **Limitation:** Lacks failover; traffic is dropped if no local endpoint exists.
+
+Note that while the initial proposal of InternalTrafficPolicy proposed a
+`PreferLocal` policy, it was dropped later on. This meant that now
+TopologyAwareRouting in conjunction with InternalTrafficPolicy didn’t exactly
+allow users to express a much desired use case from topologyKeys which is
+"prefer node-local, failover to same zone, then route anywhere" While this
+specific behavior is a non-goal for the initial implementation of the
+`routingPreference` field, the design allows for the potential introduction of
+such a preference in future refinements.
+
+## Goals
+
+* **Enhanced User Control:** Provide users with a knob to specify a preference that
+  MAY influence the traffic routing of a Kubernetes Service.
+
+* **Guidance for Implementations:** Offer standard routing preference values
+  that implementations can choose to recognize and support, promoting a degree
+  of commonality.
+
+* **Flexibility:** Allow implementations to interpret and support standard
+  heuristics in a manner that aligns with their capabilities and design.
+
+* **Extensibility:** Enable Kubernetes implementations to introduce and support
+  innovative routing strategies (for example, those based on topology, latency,
+  or other custom heuristics) that can evolve alongside users needs and
+  infrastructure capabilities.
+
+## Non-Goals
+
+* **Strict Routing Guarantees:** The `routingPreference` field will not
+  enforce deterministic routing paths. It serves as a mechanism for expressing
+  hints and preferences that implementations can consider when making routing
+  decisions.
```

Comment:
If the guarantees aren't there, are we as a side effect making it harder for Kubernetes to offer those guarantees one day?